plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java

```diff
@@ -1384,6 +1387,85 @@ public void testRowLevelDelete()
                 .hasStackTraceContaining("Delete without primary key or partition key is not supported");
     }
 
+    // test polymorphic table function
+
+    @Test
+    public void testNativeQuerySelectFromNation()
+    {
+        assertQuery(
+                "SELECT * FROM TABLE(cassandra.system.query(query => 'SELECT name FROM tpch.nation WHERE nationkey = 0 ALLOW FILTERING'))",
+                "VALUES 'ALGERIA'");
+        assertQuery(
+                "SELECT name FROM TABLE(cassandra.system.query(query => 'SELECT * FROM tpch.nation')) WHERE nationkey = 0",
+                "VALUES 'ALGERIA'");
+        assertThat(query("SELECT * FROM TABLE(cassandra.system.query(query => 'SELECT * FROM tpch.nation')) WHERE nationkey = 0"))
+                .isNotFullyPushedDown(FilterNode.class);
+    }
+
+    @Test
+    public void testNativeQuerySelectFromTestTable()
+    {
+        String tableName = "tpch.test_select" + randomNameSuffix();
+        onCassandra("CREATE TABLE " + tableName + "(col BIGINT PRIMARY KEY)");
+
+        onCassandra("INSERT INTO " + tableName + "(col) VALUES (1)");
```

Comment:
Can we use `testTable` so we don't have to have a dedicated query executor for cassandra  and we don't have to drop the table.

Reply:
Note that `TestCassandraTable` doesn't support single column with primary key.

Reply:
In this case can we move dropping operations to `finally` block ?

Reply:
Since we use `onCassandra` at a some places - Can we add support for `TestCassandraTable` with a single column primary key ?

Reply:
Let me handle in follow-up PR.